source/kraken/realtime.cpp

```diff
@@ -635,6 +635,10 @@ void handle_realtime(const std::string& id,
             meta_vj = data.pt_data->meta_vjs.emplace(trip_update.trip().trip_id());
             // TODO : pick a meaningful TZ
             meta_vj->tz_handler = data.pt_data->tz_manager.get_first_timezone();
+        } else if (is_cancelled_trip(trip_update)) {
```

Comment:
I don't think this should be specific case.
We want to track it all the same, whether it's deletion or modification on a nonexistent trip, no?

Reply:
I think you've right. 
We must to reject all COTS if it is **not a added trip** and the **Meta VJ doesn't exist**. 
This line just allows to have a specific log for the cancelled case, but it is not critic. 
I don't care, we can suppress it if you want. 

The most important thing is to reject bad case. :100:

Reply:
I think that we all agree that this check is redundant with the following one. Although, I agree with Benoit that it would be interesting to understand where this tip_update comes from (for debugging purposes).

As Kirin should already do the check of the existing VJ, we might want to **warn** about it. And to be able to track back which VJ we are working with, we need the log the timestamp/effect.

```cpp
} else {
            LOG4CPLUS_WARN(log, "Cannot perform operation on an unknown Meta VJ (other than adding trip)"
                    << ", trip id: " << trip_update.trip().trip_id() 
                    << ", effect: " << Alert_Effect_Name(trip_update.GetExtension(kirin::effect)));
            if(trip_update.stop_time_update_size()) {
                LOG4CPLUS_WARN(log, "Meta VJ 1sr stop time departure: "
                    << trip_update.stop_time_update(0).departure().time());
            }
            return;
}
```